java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java

```diff
@@ -0,0 +1,113 @@
+package com.esri.arcgisruntime.sample.rasterlayergeopackage;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.data.GeoPackage;
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.Raster;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // inflate MapView from layout
+    mMapView = findViewById(R.id.mapView);
+
+    // create a map with the BasemapType topographic
+    ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS, 39.7294, -104.8319, 11);
+
+    // set the map to be displayed in this view
+    mMapView.setMap(map);
+
+    requestReadPermission();
+  }
+
+  private void rasterLayerGeoPackage() {
+
+    // open the GeoPackage
+    GeoPackage geoPackage = new GeoPackage(
+        Environment.getExternalStorageDirectory() + getString(R.string.geopackage_path));
+    geoPackage.loadAsync();
+    geoPackage.addDoneLoadingListener(() -> {
+      if (geoPackage.getLoadStatus() == LoadStatus.LOADED) {
+        // read raster images and get the first one
+        Raster geoPackageRaster = geoPackage.getGeoPackageRasters().get(0);
+
+        // make sure an image was found in the package
```

Comment:
If there's no image in the package, I'd expect an exception will already have been thrown by this point - as the raster at index 0 has already been accessed in the line above.

Reply:
Hmm, good point...